api/api-artifacts-v1/src/main/java/com/thoughtworks/go/apiv1/artifacts/represernter/ArtifactConfigRepresenter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.artifacts.represernter;
+
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.api.representers.ErrorGetter;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.config.*;
+import com.thoughtworks.go.spark.Routes;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.thoughtworks.go.CurrentGoCDVersion.apiDocsUrl;
+
+public class ArtifactConfigRepresenter {
+    public static void toJSON(OutputWriter outputWriter, ArtifactConfig artifacts) {
+
+        Double purgeStartDiskSpace = artifacts.getPurgeSettings().getPurgeStart().getPurgeStartDiskSpace();
+        Double purgeUptoDiskSpace = artifacts.getPurgeSettings().getPurgeUpto().getPurgeUptoDiskSpace();
+        String artifactsDir = artifacts.getArtifactsDir().getArtifactDir();
+
+        outputWriter.addLinks(outputLinkWriter -> outputLinkWriter
+                .addAbsoluteLink("doc", apiDocsUrl("#artifacts"))
+                .addLink("self", Routes.Artifacts.BASE))
+                .add("artifacts_dir", artifactsDir)
+                .addChild("purge_settings", purgeSettingWriter -> {
+                    if (purgeStartDiskSpace != null) {
+                        purgeSettingWriter.add("purge_start_disk_space", purgeStartDiskSpace);
+                    }
+                    if (purgeUptoDiskSpace != null) {
+                        purgeSettingWriter.add("purge_upto_disk_space", purgeUptoDiskSpace);
+                    }
+                });
```

Comment:
If both `purgeStartDiskSpace` and `purgeUptoDiskSpace` are null, this will render a empty child, which does not seem right. 
Suggestion: delegate this to a `PurgeSettingsRepresenter` which will check for null and ensure that the `purge_settings` is only added if both are present